climate_emotions_map/make_map.py

```diff
@@ -0,0 +1,277 @@
+        data=np.array([states, single_states, abbrevs]).T,
+        columns=["state", "single_state", "state_abbreviated"],
+    )
+    return state_abbrevs_long
+
+
+def make_map(
+    question: str,
+    sub_question: int,
+    outcome: str,
+    clicked_state: str | None = None,
+    impact: str | None = None,
+    show_impact_as_gradient=True,
+    opinion_colormap: str | None = None,
+    impact_colormap: str | None = "OrRd",
+    clicked_state_marker: dict | None = None,
+    impact_marker_size_scale: float = 1.0,
+) -> go.Figure:
+    """Generate choropleth map showing opinion and/or impact data.
+
+    Parameters
+    ----------
+    question : str
+        The question to plot (for opinion data).
+    sub_question : int
+        The subquestion to plot (for opinion data).
+    outcome : str
+        The outcome to plot (for opinion data).
+    clicked_state : str | None, optional
+        Clicked state to highlight, by default None.
+    impact : str | None, optional
+        Name of impact to plot, by default None
+    show_impact_as_gradient : bool, optional
+        Whether to show impact information in the base map (replacing opinion
+        data) instead of as an additional scatter plot, by default True
+    opinion_colormap : str | None, optional
+        Colormap for the opinion data, by default None
+    impact_colormap : str | None, optional
+        Colormap for the impact data, by default "OrRd"
+    clicked_state_marker : dict | None, optional
+        Configuration for the clicked state (e.g., highlighting color). By
+        default it will make the outline thicker and yellow.
+    impact_marker_size_scale : float, optional
+        Scale factor for the impact marker size, by default 1.0
+
+    Returns
+    -------
+    go.Figure
+
+    Raises
+    ------
+    RuntimeError
+        If the given question/subquestion/outcome or impact are invalid.
+    """
+    # constants
+    col_location = "state"
+    col_color = "percentage"
+    col_color_opinion = f"{col_color} (opinion)"
+    col_color_impact = f"{col_color} (impact)"
+
+    # default values
+    if clicked_state_marker is None:
+        clicked_state_marker = dict(line=dict(width=4, color="yellow"))
+
+    # get the state abbreviations in long format
+    # "state" (i.e. state or cluster), "single_state", "state_abbreviated"
+    state_abbrevs_long = get_state_abbrevs_in_long_format()
+
+    # get the question data
+    df_opinions = opinions_state.loc[
+        (opinions_state["question"] == question)
+        & (opinions_state["sub_question"] == str(sub_question))
+        & (opinions_state["outcome"] == outcome)
+    ]
+
+    if len(df_opinions) == 0:
+        raise RuntimeError(
+            f"No data found for question {question} ({type(question)})"
+            f", sub_question {sub_question} ({type(sub_question)})"
+            f", outcome {outcome} ({type(outcome)})"
+        )
+
+    df_to_plot = df_opinions.rename(columns={col_color: col_color_opinion})
```

Comment:
```suggestion
    # We have to rename the column `col_color` (e.g. "percentage") with a suffix for `opinion`
    # because we later merge `opinion` and `impact` data on `col_color` in the same table
    # and we want to differentiate between the two
    df_to_plot = df_opinions.rename(columns={col_color: col_color_opinion})
```